core-bundle/src/Controller/Page/AbstractLayoutPageController.php

```diff
@@ -44,6 +46,17 @@ public function __invoke(Request $request): Response
             throw $this->createNotFoundException();
         }
 
+        // Override the locale based on the page's language
+        $locale = LocaleUtil::formatAsLocale($page->language);
+        $this->container->get('translator')->setLocale($locale);
+        $request->setLocale($locale);
```

Comment:
Shouldn't we use the LocaleSwitcher?🧐

Reply:
I think changing the locale of the request is correct here as every listener that follows should also have the locale of the page. 
But as @fritzmg wrote in the initial comment, I think 
> …this should actually be done earlier by the routing…